api/envoy/config/filter/http/aws_lambda/v2alpha/aws_lambda.proto

```diff
@@ -20,13 +20,27 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // AWS Lambda filter config
 message Config {
+  enum InvocationMode {
```

Comment:
Let's make this `oneof` and add the validation that it's required.

Reply:
why?

Reply:
oneof is a union. I don't need a union here. Is that an Envoy recommendation?

Reply:
One big benefit of the union is that if we ever need config that is specific to a given mode, then we'd put it as fields in those respective union variants.

A pattern I see is:

```
message SomeThingConfig {
    SomethingCommonConfig  common = 0;
    oneof kind {
        SomethingAConfig a = 1;
        SomethingBConfig b = 2;
    }
}
```